core/arch/arm32/include/kernel/thread.h

```diff
@@ -47,12 +49,15 @@ struct thread_smc_args {
 	uint32_t a6;	/* Not used */
 	uint32_t a7;	/* Hypervisor Client ID */
 };
+#endif /*ARM32*/
```

Comment:
I'm having the ARM64 version of this struct just below this struct in the ARM64 patch. What I'm aiming for is available at https://github.com/jenswi-linaro/optee_os/tree/a64_wip.

I guess it's a matter of taste if the different versions of the structs should be close to each other or if the ARM32 structs should be in one ifdef and then the ARM64 in another ifdef.